tests/std/syntax/e2equery.test.luau

```diff
@@ -0,0 +1,59 @@
+local test = require("@std/test")
+local transforms = require("./e2eCases")
+local process = require("@lute/process")
+local path = require("@std/path")
+local fs = require("@std/fs")
+
+local function stylua(...: string)
+	process.run({ "stylua", ... }, {
+		shell = true,
+	})
+end
+
+local lutePath = process.execpath()
+
+local allCasesPath = path.format(path.join(".", "tests", "std", "syntax", "e2eCases"))
+
+local function formattedTransform(initialPath: string, expectedPath: string, transformPath: string)
+	stylua(initialPath, expectedPath)
+	process.run({ lutePath, "transform", transformPath, initialPath })
+	stylua(initialPath)
+end
+
+test.suite("Query and Transform e2e", function(suite)
+	for k, _ in transforms do
+		local transformName = k
+		local transformPath = path.format(path.join(allCasesPath, transformName, "init.luau"))
+		local transformCasesPath = path.format(path.join(allCasesPath, transformName, "cases"))
+
+		assert(fs.exists(transformPath), `{transformPath} does not exist.`)
+
+		suite:case(transformName, function(assert)
+			for i, v in fs.listdirectory(transformCasesPath) do
+				if v.type ~= "dir" then
+					continue
+				end
+
+				local subcase = v.name
+				local subcaseBasePath = path.format(path.join(transformCasesPath, subcase))
+				local initialPath = path.format(path.join(subcaseBasePath, "initial.luau"))
+				local expectedPath = path.format(path.join(subcaseBasePath, "expected.luau"))
+				local resultPath = path.format(path.join(subcaseBasePath, "result.luau"))
+
+				local initialSrc = fs.readfiletostring(initialPath)
+
+				formattedTransform(initialPath, expectedPath, transformPath)
+
+				local result = fs.readfiletostring(initialPath)
+				fs.writestringtofile(resultPath, result)
+				fs.writestringtofile(initialPath, initialSrc)
+
+				local expectedResult = fs.readfiletostring(expectedPath)
+
+				assert.eq(result, expectedResult)
```

Comment:
1. We already spoke offline about changing the structure here.
2. You should think about what happens if a given test _doesn't_ conform to the expected structure.